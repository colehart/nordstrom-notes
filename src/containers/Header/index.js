import React from 'react';
import './Header.css';

export const Header = () => {
  return (
    <header className='Header'>
      Header
    </header>
  )
}