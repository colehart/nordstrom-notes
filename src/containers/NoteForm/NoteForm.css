.NoteForm {
  height: 100%;
  width: 100%;
}

.nf-form {
  width: 100%;
  max-width: 350px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.nf-note-text {
  width: 100%;
  min-height: 150px;
  padding: 15px;
  resize: none;
  outline: none;
  border: 2px solid var(--medium-gray);
}

.nf-note-text,
.nf-note-text::placeholder {
  font-family: 'Josefin Sans', sans-serif;
  font-size: 1rem;
}

.nf-note-text:valid {
  border: 2px solid var(--gray);
}

.nf-note-text,
.nf-note-text:valid {
  transition: 0.5s;
}

.nf-button-group {
  height: 115px;
  margin-top: 40px;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}

.nf-dropdown-group {
  display: grid;
  grid-template-rows: 1fr;
  grid-template-columns: 45px 1fr;
  justify-content: space-between;
  align-items: center;
  transition: 0.3s;
}

.nf-dropdown-group label {
  grid-column: 1 / 2;
}

.nf-dropdown {
  grid-column: 2 / 3;
}

.nf-dropdown-group,
.nf-submit {
  width: 100%;
}

.nf-btn {
  height: 40px;
  font-size: 1rem;
  font-family: 'Josefin Sans', sans-serif;
  border-radius: 0;
  outline: none;
  line-height: 2.45;
}

.nf-submit {
  background-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
}

.nf-submit,
.nf-submit:disabled,
.nf-submit:disabled:hover {
  color: white;
  border: 2px solid transparent;
}

.nf-submit:hover {
  background-color: white;
  border: 2px solid black;
  color: black;
  font-weight: 600;
}

.nf-submit:active {
  background-color: var(--lt-gray);
}

.nf-submit:disabled {
  background-color: var(--gray);
}

.nf-submit:disabled:hover {
  font-weight: 400;
}

.nf-submit,
.nf-submit:active,
.nf-submit:hover,
.nf-submit:disabled {
  transition: 0.3s;
}

@media screen and (max-width: 890px) {
  .nf-note-text,
  .nf-button-group {
    max-width: 292px;
  }
}

@media screen and (max-width: 500px) {
  .nf-note-text,
  .nf-button-group {
    max-width: 224px;
  }
}