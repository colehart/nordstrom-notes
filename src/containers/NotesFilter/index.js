import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { caughtError } from '../../actions';
// import { filterNotes } from '../../actions';
import './NotesFilter.css';

export class NotesFilter extends Component {
  constructor () {
    super()
    this.state = {
      filter: '',
      filterParams: '',
      isDisabled: true,
      placeholderText: 'Choose a filter.',
    }
  }

  handleInputChange = async event => {
    const { name, value } = event.target;

    await this.setState({ [name]: value })
    if (name === 'filter') this.toggleParams()
  }

  toggleParams = async () => {
    const { filter } = this.state

    filter
      ? await this.setState({ isDisabled: false })
      : await this.setState({ isDisabled: true })

    this.formatPlaceholder(filter)
  }

  formatPlaceholder = async filter => {
    const dateState = { placeholderText: 'Ex: 12-18-2018' }
    const tagState = { placeholderText: 'Ex: Work, Hobby...' }

    filter === 'date'
      ? await this.setState(dateState)
      : await this.setState(tagState)
  }

  render() {
    const { filter, filterParams, isDisabled, placeholderText } = this.state;

    return (
      <section className='NotesFilter'>
        <form
          className='nl-form'
          // onSubmit={this.props.filterNotes()}
        >
          <div className='nl-filter-by'>
            <label htmlFor='filter-group'>Filter by:</label>
            <div
              className="nl-filter-group"
              id='filter-group'
              name='filter-type'
            >
              <div className='nl-filter-type'>
                <input
                  type='radio'
                  name='filter'
                  value='date'
                  id='date'
                  onChange={this.handleInputChange}
                />
                <label htmlFor="date">Date</label>
              </div>
              <div className='nl-filter-type'>
                <input
                  type='radio'
                  name='filter'
                  value='tag'
                  id='tag'
                  onChange={this.handleInputChange}
                />
                <label htmlFor='tag'>Tag</label>
              </div>
            </div>
          </div>
          <div className='nl-filter-params-group'>
            <label htmlFor='filter-params'>Search:</label>
            <input
              className='nl-filter-params'
              id='filter-params'
              name='filterParams'
              value={filterParams}
              maxLength={ filter === 'date' ? '10' : '8' }
              onChange={this.handleInputChange}
              disabled={isDisabled}
              placeholder={placeholderText}
            />
          </div>
        </form>
      </section>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  caughtError: (errorMessage) => dispatch(caughtError(errorMessage)),
  // filterNotes: () => dispatch(filterNotes()),
})

NotesFilter.propTypes = {
  caughtError: PropTypes.func,
  filterNotes: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(NotesFilter)