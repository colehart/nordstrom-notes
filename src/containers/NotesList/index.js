import React from 'react';
import './NotesList.css';

export const NotesList = () => {
  return (
    <div className='NotesList'>
    </div>
  )
}