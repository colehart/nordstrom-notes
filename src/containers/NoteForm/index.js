import React, { Component } from 'react';
import TextareaAutosize from 'react-autosize-textarea';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';
import { addNotes, caughtError, newNoteAdded } from '../../actions';
import './NoteForm.css';


export class NoteForm extends Component {
  constructor() {
    super()
    this.state = {
      tag: 'Work',
      text: '',
      isDisabled: true,
      wasSuccessful: false,
      timoutId: null,
    }
  }

  componentWillUnmount() {
    const { timeoutId } = this.state
    clearTimeout(timeoutId);
  }

  handleInputChange = async event => {
    const { name, value } = event.target;

    await this.setState({ [name]: value })
    if (name === 'text') this.toggleSubmit();
  }

  toggleSubmit = async () => {
    const { text } = this.state

    text
      ? await this.setState({ isDisabled: false })
      : await this.setState({ isDisabled: true })
  }

  handleSubmit = async event => {
    event.preventDefault();
    const { tag, text } = this.state
    const date = moment().format('MM-DD-YYYY')

    try {
      const successState = {
        tag: 'Work',
        text: '',
        isDisabled: true,
        wasSuccessful: true,
      }

      await this.props.addNotes(tag, text, date)
      await this.props.newNoteAdded(true)
      await this.setState(successState)
      this.resetBanner()
    }
    catch(error) {
      this.props.caughtError(error.message)
    }
  }

  resetBanner = async () => {
    const id = setTimeout(() => this.setState({ wasSuccessful: false }), 1000)
    await this.setState({ timeoutId: id })
  }

  render() {
    const { text, isDisabled, wasSuccessful } = this.state

    return (
      <section className='NoteForm'>
        <div className={ wasSuccessful ? 'nf-success' : 'hidden' }>
          Note successfully added!
        </div>
        <h2>Make Note</h2>
        <form
          className='nf-form'
          onSubmit={this.handleSubmit}
        >
          <TextareaAutosize
            className='nf-note-text'
            name='text'
            value={text}
            onChange={this.handleInputChange}
            placeholder='Write your thoughts here...'
            aria-label='Write your note here, in 250 characters or less'
            maxLength='250'
            wrap='soft'
            required
            onResize={(e) => {}}
          />
          <div className="nf-button-group">
            <div className="nf-dropdown-group">
              <label htmlFor='nf-tags'>Tag:</label>
              <select
                className="nf-dropdown nf-btn"
                name='tag'
                onChange={this.handleInputChange}
                id='nf-tags'
              >
                <option value='Work'>Work</option>
                <option value='Personal'>Personal</option>
                <option value='Hobby'>Hobby</option>
              </select>
            </div>
            <button
              className='nf-submit nf-btn'
              disabled={isDisabled}
              type='submit'
              onClick={this.handleSubmit}
            >
              Make Note
            </button>
          </div>
        </form>
      </section>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addNotes: (tag, text, date) => dispatch(addNotes(tag, text, date)),
  caughtError: (errorMessage) => dispatch(caughtError(errorMessage)),
  newNoteAdded: (bool) => dispatch(newNoteAdded(bool))
})

NoteForm.propTypes = {
  addNotes: PropTypes.func,
  caughtError: PropTypes.func,
  newNoteAdded: PropTypes.func,
}

export default connect(null, mapDispatchToProps)(NoteForm)