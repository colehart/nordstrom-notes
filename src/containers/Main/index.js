import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import NoteForm from '../../containers/NoteForm';
import { NotesList } from '../../containers/NotesList';
import './Main.css';

export const Main = ({ isLoading, match }) => {
  return (
    <main className='Main'>
      { isLoading ? <Loader /> : '' }
      { match.path === '/'
          ? <NoteForm />
          : <NotesList />
      }
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

Main.propTypes = {
  isLoading: PropTypes.bool
}

export default withRouter(connect(mapStateToProps)(Main))
