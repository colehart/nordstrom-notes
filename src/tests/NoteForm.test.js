import React from 'react';
import { shallow } from 'enzyme';
import { NoteForm } from '../containers/NoteForm';

describe('NoteForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NoteForm />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})