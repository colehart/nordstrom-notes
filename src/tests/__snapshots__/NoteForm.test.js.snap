// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm matches the snapshot 1`] = `
<section
  className="NoteForm"
>
  NoteForm
</section>
`;
